auraed/src/runtime/mod.rs

```diff
@@ -56,58 +57,88 @@ pub struct CoreService {}
     async fn run_executable(
         &self,
         request: Request<Executable>,
     ) -> Result<Response<ExecutableStatus>, Status> {
         let r = request.into_inner();
-        let cmd = command_from_string(&r.command);
-        match cmd {
-            Ok(mut cmd) => {
-                let output = cmd.output();
-                match output {
-                    Ok(output) => {
-                        let meta = meta::AuraeMeta {
-                            name: r.command,
-                            message: "-".to_string(),
-                        };
-                        let proc = meta::ProcessMeta { pid: -1 }; // todo @kris-nova get pid, we will probably want to spawn() and wait and remember the pid
-                        let status = meta::Status::Complete as i32;
-                        let response = ExecutableStatus {
-                            meta: Some(meta),
-                            proc: Some(proc),
-                            status,
-                            stdout: String::from_utf8(output.stdout)
-                                .expect("reading stdout"),
-                            stderr: String::from_utf8(output.stderr)
-                                .expect("reading stderr"),
-                            exit_code: output.status.to_string(),
-                        };
-                        Ok(Response::new(response))
-                    }
-                    Err(e) => {
-                        let meta = meta::AuraeMeta {
-                            name: "-".to_string(),
-                            message: format!("{:?}", e),
-                        };
-                        let proc = meta::ProcessMeta { pid: -1 };
-                        let status = meta::Status::Error as i32;
-                        let response = ExecutableStatus {
-                            meta: Some(meta),
-                            proc: Some(proc),
-                            status,
-                            stdout: "-".to_string(),
-                            stderr: "-".to_string(),
-                            exit_code: "-".to_string(),
-                        };
-                        Ok(Response::new(response))
-                    }
+        // let rmeta = r.meta.expect("parsing request meta");
+
+        // Build command to execute.
+        let mut cmd =
+            command_from_string(&r.command).expect("command string parsing");
+
+        // Calculate the ID for the cell to run the command in
+        let cell_id = cell_name_from_string(&r.command)
+            .expect("cell name hash calculation");
+
+        info!("Spawning cell: {}", cell_id);
+
+        // Create the cgroup for the cell
+        let hierarchy = cgroups_rs::hierarchies::auto(); // v1/v2 cgroup switch automatically
+        let cgroup: Cgroup = CgroupBuilder::new(&cell_id)
+            .cpu()
+            .shares(10) // Use 10% of the CPU relative to other cgroups
```

Comment:
should this be part of the Executable Request?

i imagine memory could be too (and maybe networking?)

Reply:
having it part of the Request would also make it easier to build a scheduling layer on top, if the "available CPU" is exposed for a Cell